### PR TITLE
tests: posix: Function call return values need to be validated.

### LINKED ISSUE #7084: [Coverity CID: 185280] Error handling issues in /tests/posix/pthread_rwlock/src/posix_rwlock.c
Static code scan issues seen in File: /tests/posix/pthread_rwlock/src/posix_rwlock.c
 Category: Error handling issues
 Function: test_rw_lock
 Component: Tests
 CID: 185280
 Please fix or provide comments to square it off in coverity in the link: https://scan9.coverity.com/reports.htm#v32951/p12996

### PR DESCRIPTION
**Pull request by sritej20**
In this fix we are asserting the return value of pthread_attr_init.

Resolves #7084
Coverity-CID: 185280

Signed-off-by: Sritej Kanakadandi Venkata Rama <sritej.kvr@gmail.com>

### CODE REVIEW

**Comment by sritej20** [comment]
@nashif  I needed to add the newline to satisfy the checkpatch.pl requirement of 80 chars per line.

**Comment by sritej20** [comment]
@andrewboie @nashif @lpereira  Appreciate another final review to get the PR in.

File: tests/posix/pthread_rwlock/src/posix_rwlock.c
Comment (andrewboie) [inline tests/posix/pthread_rwlock/src/posix_rwlock.c]: this should be aligned with the opening parenthesis of zassert_equal(), not way out over here

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]

**Comment by sritej20** [comment]
@andrewboie Updated the patch

**Review by nashif — APPROVED** [review]

**Comment by sritej20** [comment]
@andrewboie i guess we need a approval from you before the changes can be merged.

**Comment by nashif** [comment]
> @nashif I needed to add the newline to satisfy the checkpatch.pl requirement of 80 chars per line.

not sure what you mean here, I am referring to "\n" at the end of the message.

**Comment by sritej20** [comment]
@nashif  My bad , Fixed that as well .

**Review by nashif — APPROVED** [review]

File: tests/posix/pthread_rwlock/src/posix_rwlock.c
Comment (sritej20) [reply tests/posix/pthread_rwlock/src/posix_rwlock.c]: Updated the patch . Please review .

**Pull request merged**
